src/meta/src/stream/scale.rs

```diff
@@ -39,9 +39,54 @@ use uuid::Uuid;
 use crate::barrier::{Command, Reschedule};
 use crate::manager::{IdCategory, WorkerId};
 use crate::model::{ActorId, DispatcherId, FragmentId, TableFragments};
-use crate::storage::MetaStore;
+use crate::storage::{MetaStore, MetaStoreError, Transaction, DEFAULT_COLUMN_FAMILY};
 use crate::stream::GlobalStreamManager;
-use crate::MetaResult;
+use crate::{MetaError, MetaResult};
+
+#[derive(Copy, Clone, Debug)]
+pub struct RescheduleRevision(u64);
+
+const RESCHEDULE_VERSION_KEY: &[u8] = b"reschedule_version";
+
+impl From<RescheduleRevision> for u64 {
+    fn from(value: RescheduleRevision) -> Self {
+        value.0
+    }
+}
+
+impl RescheduleRevision {
+    pub async fn get<S>(store: &S) -> MetaResult<Self>
+    where
+        S: MetaStore,
+    {
+        let version = match store
+            .get_cf(DEFAULT_COLUMN_FAMILY, RESCHEDULE_VERSION_KEY)
+            .await
+        {
+            Ok(byte_vec) => memcomparable::from_slice(&byte_vec).unwrap(),
+            Err(MetaStoreError::ItemNotFound(_)) => 0,
+            Err(e) => return Err(MetaError::from(e)),
+        };
+
+        Ok(Self(version))
+    }
+
+    pub fn increase(&self) -> Self {
```

Comment:
Do we also need to increase the revision on migration / fail-over?

Reply:
👍 I think we might need to increate the revision as well in these scenarios:
1. optional for drop streaming jobs, if not `reschedule` will return `fragment not found` error. We'd better use `revision mismatch` to unify it.
2. might needed for table schema change? I'm not sure @BugenZhao .
3. needed for migration, but not for failover.

Reply:
Table schema change will build new fragments with new fragment IDs, which won't make the rescheduling plan ambiguous as long as the user does not refer to an inexistent fragment ID deliberately. 😂

Reply:
Ideally, we can cover all kinds of changes to the streaming graph with revision, but it might be hard to implement or refactor. So only considering the parts that matter to scaling LGTM.

Reply:
Indeed, I actually prefer having a global version to represent the current meta revision. I found that [SnapshotVersion](https://github.com/risingwavelabs/risingwave/blob/3bb437ba7fd58c14490e6472978acd74d84a7f39/proto/meta.proto#L241) is quite suitable, but it seems to be for the Notification service. It feels like we can refactor in the future, introducing a similar global revision for meta.

Reply:
updated